make.js

```diff
@@ -244,9 +236,28 @@ CLI.serverBuild = function(/** @type {{ node: string; task: string }} */ argv) {
     banner('Build successful', true);
 }
 
-function buildTask(taskName, taskListLength) {
+function getNodeVersion (buildPath, taskName) {
+    var taskJsonPath = path.join(buildPath, taskName, "task.json");
+    if (!fs.existsSync(taskJsonPath)) {
+        console.warn('Unable to find task.json, defaulting to use Node 10');
+        return 10;
+    }
+    var taskJsonContents = fs.readFileSync(taskJsonPath, { encoding: 'utf-8' });
+    var taskJson = JSON.parse(taskJsonContents);
+    var execution = taskJson['execution'] || taskJson['prejobexecution'];
+    var nodeVersion = 10;
+    for (var key of Object.keys(execution)) {
+        const executor = key.toLocaleLowerCase();
```

Comment:
Some thoughts, the logic won't work if you'll put the Node20 executor on top of the execution object because it will be overridden by the executor below, example: 
```
"execution": {
  "Node20": {
    "target": "sometask.js",
    "argumentFormat": ""
  },
  "Node16": {
    "target": "sometask.js",
    "argumentFormat": ""
  }
}, 
 ``` 
 will return 16 instead of 20, it probably needs some sorting to return the highest version of the node version?

Reply:
Thanks for catching it, updated.